### PR TITLE
adding 2 tests & apply eslint to the tests

### PR DESCRIPTION
**Pull request by hasezoey**
adding tests for:
- correctly save parameters, without giving "previousHash"
- BlockChain.getAllTransactionsForWallet

bringing coverage to about 99%

adding eslint to the tests

### CODE REVIEW

File: tests/blockchain.test.js [hunk 1]
```diff
@@ -88,12 +88,32 @@ describe('Blockchain class', function() {
...
+			const validTx = createSignedTx();
+    		blockchain.addTransaction(validTx);
+    		blockchain.addTransaction(validTx);
+
+			blockchain.minePendingTransactions("b2");
+			blockchain.addTransaction(validTx);
+			blockchain.addTransaction(validTx);
+			blockchain.minePendingTransactions("b2");
+
+			blockchain.getAllTransactionsForWallet("b2").forEach((x) => {
```
Comment (Savjee) [inline tests/blockchain.test.js]: Could you refactor this with a regular for loop?
As in: `for(const tx of blockchain.getAllTransactionsForWallet())`

Comment (hasezoey) [reply tests/blockchain.test.js]: yes will make

File: tests/blockchain.test.js [hunk 2]
```diff
@@ -96,12 +96,32 @@ describe('Blockchain class', function() {
...
     		blockchain.chain[1].timestamp = 897397;
     		assert(!blockchain.isChainValid());
     	});
-    });
+	});
+	
+	describe("getAllTransactionsForWallet", function(){
+		it("should get all Transactions for a Wallet", function() {
+			const validTx = createSignedTx();
+    		blockchain.addTransaction(validTx);
```
Comment (Savjee) [inline tests/blockchain.test.js]: Can you fix the indentation here?

File: tests/blockchain.test.js [hunk 3]
```diff
@@ -96,12 +96,32 @@ describe('Blockchain class', function() {
...
+			blockchain.minePendingTransactions("b2");
+			blockchain.addTransaction(validTx);
+			blockchain.addTransaction(validTx);
+			blockchain.minePendingTransactions("b2");
+
+			for (const trans of blockchain.getAllTransactionsForWallet("b2")) {
+				assert.equal(trans.amount, 100);
+				assert.equal(trans.fromAddress, null);
+				assert.equal(trans.toAddress, "b2");
+				// timestamp cant be verified?
```
Comment (Savjee) [inline tests/blockchain.test.js]: No it can't. You can remove this comment ;)

File: tests/blockchain.test.js [hunk 4]
```diff
@@ -96,12 +96,32 @@ describe('Blockchain class', function() {
...
+			const validTx = createSignedTx();
+    		blockchain.addTransaction(validTx);
+    		blockchain.addTransaction(validTx);
+
+			blockchain.minePendingTransactions("b2");
+			blockchain.addTransaction(validTx);
+			blockchain.addTransaction(validTx);
+			blockchain.minePendingTransactions("b2");
+
+			for (const trans of blockchain.getAllTransactionsForWallet("b2")) {
```
Comment (Savjee) [inline tests/blockchain.test.js]: I guess we should also check the length of the array that is returned by `blockchain.getAllTransactionsForWallet("b2")`. It should only contain 2 transactions.

File: tests/blockchain.test.js (hunk 2 above)
Comment (hasezoey) [reply tests/blockchain.test.js]: i dont know how this happened ...

File: tests/blockchain.test.js (hunk 4 above)
Comment (hasezoey) [reply tests/blockchain.test.js]: ok will add

File: tests/blockchain.test.js (hunk 2 above)
Comment (hasezoey) [reply tests/blockchain.test.js]: fixed with adding eslint to the tests

**Comment by hasezoey** [comment]
hope its ok to add eslint to the tests, to not confuse anybody working with the tests & prevent random indent changes

**Comment by Savjee** [comment]
Thanks again for your hard work! Just one more remark: could you limit each PR to just 1 change?
Here you implemented a few new tests, redid the formatting and added eslint to all tests.

**Comment by hasezoey** [comment]
yes, ok, i needed to add eslint here, because at my end, the indent was right (and it was at up-to-date)

**Pull request merged**